### PR TITLE
Merge 4 Util test classes

### PR DESCRIPTION
**Pull request by sinasen2000**
Possible changes for Misc class:
- :  Changing the naming of listToString method to stringListToString, or even better, add support for other types such as integers.
- : In cleanseMessage, the parameter "hideRegex" should be renamed. It can be understood as a boolean to hide the regex portions in the string or not. A possible renaming can be "regexForCleaning" or "regexForHiding".

### CODE REVIEW

File: core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     *
+     * Method: performAction(int action)
+     *
+     */
+    @Test
+    public void testPerformAction34() throws Exception {
+        CounterStatistic cs = new CounterStatistic(10);
+        cs.performAction(3);
+        assertEquals(cs.getValue(), 10);    }
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java:48]: code style dictates that the closing bracket, when present should be on a new line.

Applies in multiple tests

File: core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+     */
+    @Test
+    public void testSetMaxSize() throws Exception {
+        SizeLimitedVector slv = new SizeLimitedVector(10);
+        slv.setMaxSize(14);
+        assertEquals(slv.getMaxSize(), 14);
+    }
+
+
+}
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java:41]: Try to also test the function of this class. set a limit, add limit +1 messages to the vector.
Test the size
Test that the first item has been removed

File: core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java (hunk 1 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java]: changed it

File: core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java (hunk 2 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java]: If I understood what you meant correctly, now it is tested

File: core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Test
+    public void testMaxSizePassed() throws ArrayIndexOutOfBoundsException {
+        try{SizeLimitedVector slv = new SizeLimitedVector(1);
+        slv.add(13);
+        slv.add(14);
+        }
+        catch (ArrayIndexOutOfBoundsException e) {
+            fail("array capacity is passed, capacity was 1, but 2 elements got added");
+        }
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java:39]: What exactly are you testing here? this catch clause should never happen?
Maybe add some test assertions :)

File: core/src/test/java/nl/nn/adapterframework/util/FilenameUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+/**
+ * FilenameUtils Tester.
+ *
+ * @author <Sina Sen>
+ */
+public class FilenameUtilsTest {
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/FilenameUtilsTest.java:13]: This class didn't require any tests. It's a utilities class copied from apache commons io.
The only reason it's here is in order to avoid websphere compilations and runtime issues.

File: core/src/test/java/nl/nn/adapterframework/util/FilenameUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+
+
+
+    /**
+     * Method: isSystemWindows()
+     */
+    @Test
+    public void testIsSystemWindows() throws Exception {
+        assertEquals(FilenameUtils.isSystemWindows(), true);
+    }
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/FilenameUtilsTest.java:23]: You might run on windows, but many of our servers do not :) this will fail on Travis and Jenkins

File: core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java (hunk 3 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java:39]: why shouldn't the catch clause happen? The second add operation passes the size of the vector thus raises and AIOoBounds exception. I'm testing the case that if one adds more items to the vector than the size obliges, what happens next. In the original method .add there was not such a case that raises an expception so I used my own one.

File: core/src/test/java/nl/nn/adapterframework/util/FilenameUtilsTest.java (hunk 5 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/FilenameUtilsTest.java]: yes indeed, I removed this test case.

File: core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java (hunk 3 above)
Comment (nielsm5) [reply core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java:39]: The whole point of the SizeLimitedVector is that you can append as many things to it as you like but it only stores the last X amount.

File: core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    /**
+     *
+     * Method: setMaxSize(int maxSize)
+     *
+     */
+    @Test
+    public void testSetMaxSize() throws Exception {
+        SizeLimitedVector slv = new SizeLimitedVector(10);
+        slv.setMaxSize(14);
+        assertEquals(slv.getMaxSize(), 14);
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java:51]: Please also add 14+1 elements and verify more then just the setter being changed.
e.g. calling getSize or capacity to verify the contents of the vector.

File: core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+public class SizeLimitedVectorTest {
+
+
+    /**
+     *
+     * Method: add(Object o)
+     *
+     */
+    @Test
+    public void testAdd() throws ArrayIndexOutOfBoundsException {
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java:22]: You dont have to declare this RuntimeException here do you? Non of the functions are throwing any exceptions

File: core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    
+
+    /**
+     *
+     * Method: performAction(int action)
+     *
+     */
+    @Test
+    public void testPerformAction34() throws Exception {
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java:46]: Try to use more descriptive names.  testAction1, testAction2, etc. is a bit vague when you're going through the log files :)

File: core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java (hunk 7 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java:22]: Yes. I am going to remove it, thanks for pointing out.

File: core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java (hunk 8 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java:46]: Indeed, the naming I used looks terrible, I used numbers to reflect the action with number given in the CounterStatistic class, however using the action names rather than the number will be more descriptive. I am changing that.

File: core/src/test/java/nl/nn/adapterframework/util/FilenameUtilsTest.java (hunk 4 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/FilenameUtilsTest.java:13]: okay, more communication will be assured on testing new classes.

File: core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java (hunk 3 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java]: yes, I changed the test accordingly.

File: core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+     */
+    @Test
+    public void testSetMaxSize() throws Exception {
+        SizeLimitedVector slv = new SizeLimitedVector(5);
+        slv.setMaxSize(6);
+        slv.add(1); slv.add(2); slv.add(3); slv.add(4); slv.add(5); slv.add(6); slv.add(7);
+        assertEquals(slv.getMaxSize(), 6);
+        assertEquals(slv.capacity(), 10);
+        assertEquals(slv.size(), 6);
+        assertEquals(slv.get(0), 2);
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java:51]: 👍

File: core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+     *
+     * Method: performAction(int action)
+     *
+     */
+    @Test
+    public void testPerformActionMarkMainOrFull() throws Exception {
+        CounterStatistic cs = new CounterStatistic(10);
+        cs.performAction(3);
+        assertEquals(cs.getValue(), 10);
+    }
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java:50]: You are testing the actions here, instead of using the integer values, 123, could you use the more descriptive `HasStatistics` integer fields? So action 2 would become `HasStatistics.STATISTICS_ACTION_RESET`.

Maybe you could also write 2 more tests for action 3 and 4 (mark main and mark full)

File: core/src/test/java/nl/nn/adapterframework/util/MiscTest.java [hunk 11]
```diff
@@ -1,43 +1,610 @@
...
+     */
+    @Test
+    public void testFileToStringFileName() throws Exception {
+        //Misc.resourceToString()
+        assertEquals(Misc.fileToString(file.getName()), "inside the lebron file");
+    }
+    @Test
+    public void testFileToStringFileNameEndLine() throws Exception {
+        //Misc.resourceToString()
+        assertEquals(Misc.fileToString(file.getName(), " the end"), "inside the lebron file");
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/MiscTest.java:259]: You're not testing the `endOfLineString` string here?

Comment (nielsm5) [reply core/src/test/java/nl/nn/adapterframework/util/MiscTest.java:259]: I see you're testing it in another test, maybe to not confuse things leave it blank or `null` instead :)

File: core/src/test/java/nl/nn/adapterframework/util/MiscTest.java [hunk 12]
```diff
@@ -1,43 +1,610 @@
...
+    /**
+     * Method: compress(String input)
+     */
+    @Test
+    public void testCompressDecompressInput() throws Exception {
+        String s = "test";
+        byte[] compressed = Misc.compress(s);
+        String decompressed = Misc.decompressToString(compressed);
+        assertEquals(decompressed, "test");
+    }
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/MiscTest.java:378]: Maybe also test the compressed variant to validate if it has actually been compressed

File: core/src/test/java/nl/nn/adapterframework/util/MiscTest.java [hunk 13]
```diff
@@ -1,43 +1,610 @@
...
+                "\n" +
+                "<body>\n" +
+                "<div>\n" +
+                "    <h1>Example Domain</h1>\n" +
+                "    <p>This domain is for use in illustrative examples in documents. You may use this\n" +
+                "    domain in literature without prior coordination or asking for permission.</p>\n" +
+                "    <p><a href=\"https://www.iana.org/domains/example\">More information...</a></p>\n" +
+                "</div>\n" +
+                "</body>\n" +
+                "</html>\n");
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/MiscTest.java:591]: In stead of using a big concatinated string, it might be easier and more readable to use `TestFileUtils.getResource` and read the contents from a file instead

File: core/src/test/java/nl/nn/adapterframework/util/MiscTest.java [hunk 14]
```diff
@@ -1,43 +1,610 @@
...
+    }
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/MiscTest.java:602]: In stead of using a big concatinated string, it might be easier and more readable to use `TestFileUtils.getResource` and read the contents from a file instead

File: core/src/test/java/nl/nn/adapterframework/util/MiscTest.java [hunk 15]
```diff
@@ -1,43 +1,610 @@
...
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/MiscTest.java:609]: In stead of using a big concatinated string, it might be easier and more readable to use `TestFileUtils.getResource` and read the contents from a file instead

**Comment by nielsm5** [comment]
Looking good, a few more bits and bobs and it's ready to go!

File: core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java (hunk 10 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java:50]: will add.

Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java:50]: added

File: core/src/test/java/nl/nn/adapterframework/util/MiscTest.java (hunk 11 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/MiscTest.java:259]: I removed this test. It was not testing anything unique.

File: core/src/test/java/nl/nn/adapterframework/util/MiscTest.java (hunk 13 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/MiscTest.java:591]: I assumed you meant `TestFileUtils.getTestFile` which uses `getResource`. So I'm using this method.

File: core/src/test/java/nl/nn/adapterframework/util/MiscTest.java (hunk 14 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/MiscTest.java:602]: done

File: core/src/test/java/nl/nn/adapterframework/util/MiscTest.java (hunk 15 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/MiscTest.java:609]: done

File: core/src/test/java/nl/nn/adapterframework/util/MiscTest.java (hunk 12 above)
Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/MiscTest.java:379]: I didn't see this yesterday, will take a look at it now.

Comment (sinasen2000) [reply core/src/test/java/nl/nn/adapterframework/util/MiscTest.java:392]: I added a case that tests if the string is compressed. However, I did this by testing the first two indexes of the compressed byte array, which was the same for all compressed Strings, so I assumed this is a sign that it is compressed. Am I correct? Is there another way of testing if it is actually compressed? @nielsm5
